lib/service/modules/kinveyEntity.js

```diff
@@ -50,98 +48,98 @@ function kinveyEntityModule(environmentId, useBSONObjectId) {
       if (typeof entityOrId === 'object' && entityOrId !== null) {
         _entity = entityOrId;
 
-        if (!has(_entity, ['_acl'])) _entity._acl = {};
-        if (!has(_entity, ['_id'])) _entity._id = _createEntityId();
-        if (!has(_entity, ['_acl', 'creator'])) _entity._acl.creator = environmentId;
-        if (!has(_entity, ['_kmd'])) _entity._kmd = {};
-        if (!has(_entity, ['_kmd', 'ect'])) _entity._kmd.ect = currentTimeString;
-        if (!has(_entity, ['_kmd', 'lmt'])) _entity._kmd.lmt = currentTimeString;
+        if (!_entity._acl) _entity._acl = {};
+        if (!_entity._id) _entity._id = _createEntityId();
+        if (!_entity._acl.creator) _entity._acl.creator = environmentId;
+        if (!_entity._kmd) _entity._kmd = {};
+        if (!_entity._kmd.ect) _entity._kmd.ect = currentTimeString;
+        if (!_entity._kmd.lmt) _entity._kmd.lmt = currentTimeString;
       }
     }
 
     let aclFunctions = {};
 
     function getCreator() {
-      return getProp(_entity, ['_acl', 'creator']);
+      return _entity._acl.creator;
```

Comment:
Will throw a `TypeError` if `_entity._acl` is null. Applies to some other occurrences below

Reply:
No, it won't, because of line 53 above.

Reply:
Missed that 👍